src/arch/util/machine-broadcast.C

```diff
@@ -120,14 +120,20 @@ void CmiForwardMsgToPeers(int size, char *msg) {
 }
 #endif
 
+static char *_lazyCopy(char* &dst, const char *src, int size, int rankToAssign) {
```

Comment:
Maybe better named _lazyCopyMsg?

Reply:
Also, it's a bit of a semantic misnomer to me, I'd expect a function called lazy copy to defer copies to point of use.